lib/CLI/Types.hs

```diff
@@ -0,0 +1,129 @@
+module CLI.Types where
+
+import Data.Text (Text)
+import Data.Time.Compat (Day, UTCTime , Year)
+import Data.Time.Calendar.Month.Compat (Month)
+import Entry (FieldKey, FieldVisibility, EntryTag)
+import Coffer.Path (Path, EntryPath)
+
+data Command
+  = CmdView ViewOptions
+  | CmdCreate CreateOptions
+  | CmdSetField SetFieldOptions
+  | CmdDeleteField DeleteFieldOptions
+  | CmdFind FindOptions
+  | CmdRename RenameOptions
+  | CmdCopy CopyOptions
+  | CmdDelete DeleteOptions
+  | CmdTag TagOptions
+  deriving stock Show
+
+----------------------------------------------------------------------------
+-- Options
+----------------------------------------------------------------------------
+
+data ViewOptions = ViewOptions
+  { voPath :: Path
+  , voFieldName :: Maybe FieldKey
+  }
+  deriving stock Show
+
+data CreateOptions = CreateOptions
+  { coPath :: EntryPath
+  , coEdit :: Bool
+  , coForce :: Bool
+  , coTags :: [EntryTag]
```

Comment:
Why not [Data.Set](https://hackage.haskell.org/package/containers-0.6.5.1/docs/Data-Set.html)?

Reply:
I'll change this next time I rebase on `magic_rb/core`

Reply:
I've rebased on `main`, and changed `coTags` to a set.